src/modules/claims/claims.service.ts

```diff
@@ -1354,22 +1360,37 @@ export class ClaimsService {
     );
   }
 
-  /**
+    /**
+   * Verifies that credential was issued by authorized issuer
+   *
+   * @param {VerifiableCredential<RoleCredentialSubject} vc to be verified
+   */
+     async verifyVc(vc: VerifiableCredential<RoleCredentialSubject>) {
+      const issuerDID = this._signerService.did;
+      if (!await this._verifiableCredentialService.verify(vc)) {
+        throw new Error(ERROR_MESSAGES.PROOF_NOT_VERIFIED)
+      }
+       const role = vc.credentialSubject.role.namespace;
+       await this._vcIssuerVerifier.verifyIssuer(issuerDID, role);
+    }
+
+    /**
    *
    * Set the Verifier for Claim Issuance.
    *
    */
-  private _setClaimIssuerVerifier() {
-    const credentialResolver: CredentialResolver = new IpfsCredentialResolver(
-      this._signerService.provider,
-      this._didRegistry.registrySettings,
-      this._didRegistry.ipfsStore
-    );
-    const domainReader = this._domainsService.domainReader;
-    const issuerResolver = new EthersProviderIssuerResolver(domainReader);
-    this._vcIssuerVerifier = new VCIssuerVerification(
-      credentialResolver,
-      issuerResolver
-    );
-  }
+     private _setClaimIssuerVerifier() {
+      const credentialResolver: CredentialResolver = new IpfsCredentialResolver(
+        this._signerService.provider,
+        this._didRegistry.registrySettings,
+        this._didRegistry.ipfsStore
+      );
+      const domainReader = this._domainsService.domainReader;
+      const issuerResolver = new EthersProviderIssuerResolver(domainReader);
+      this._vcIssuerVerifier = new VCIssuerVerification(
+        issuerResolver,
+        credentialResolver,
+        this.verifyProof,
+      );
+    }
```

Comment:
If possible, it would be nice to have tests of the new `verifyVc` functionality. This might be a bit tricky to do because of these dependencies but I think it's at least worth trying.
For the credentialResolver, maybe we can provide a mock credentialResolver or spawn an `ipfsDeamon`. I think the latter is the approach that ew-credentials has taken https://github.com/energywebfoundation/ew-credentials/blob/bc54a588ec2a7547b6711f184035de638918c66b/packages/vc-verification/test/vc-issuer-verification.ts#L126
For having the roles in place, I think we have some tests that create roles in the `domains.service.e2e.ts`, so maybe we could use/copy that code to set up a role hierarchy

If the above is too complex/too much work and there is a simpler way, we can look at that too

My rationale for wanting tests is that
- They document the expected behaviour
- If there is a breaking change to the behaviour of dependencies (e.g. to `VCIssuerVerification.verifyIssuer`, then if we have tests in ICL, then our tests can break and we can know that it is a breaking change to ICL.

Reply:
ok. @JGiter hope you wouldn't mind if I seek some input on how you would think about going about this.

Reply:
Also looks like this PR is blocking a PR for Jakub. Not sure how urgent his is, if we need to merge this and make a task this sprint to do a test or if his can wait.